tests/unit/Framework/MockObject/GeneratorTest.php

```diff
@@ -45,6 +45,41 @@ public function testGetMockFailsWhenInvalidFunctionNameIsPassedInAsAFunctionToMo
         $this->generator->getMock(stdClass::class, [0]);
     }
 
+    public function testGetMockFailsWithInvalidClass(): void
+    {
+        $this->expectException(\PHPUnit\Framework\InvalidArgumentException::class);
+
+        $this->generator->getMock(false);
+    }
+
+    public function testGetMockFailsWithInvalidMethods(): void
+    {
+        $this->expectException(\PHPUnit\Framework\InvalidArgumentException::class);
+
+        $this->generator->getMock(stdClass::class, false);
+    }
+
+    public function testGetMockFailsWithNonExistingClass(): void
+    {
+        $this->expectException(\PHPUnit\Framework\MockObject\RuntimeException::class);
+
+        $this->generator->getMock('Tux', [], [], '', true, true, false, true, false, null, false);
+    }
+
+    public function testGetMockFailsWithNonExistingClasses(): void
+    {
+        $this->expectException(\PHPUnit\Framework\MockObject\RuntimeException::class);
+
+        $this->generator->getMock(['Tux', 'Bar'], [], [], '', true, true, false, true, false, null, false);
+    }
+
+    public function testGetMockWithExistingClassAsMockName(): void
```

Comment:
I propose to add `…throwsExcepction` to the corresponding test method names so we have a kind of sentence describing the behavior.

Reply:
done